packages/core-input/core-input.js

```diff
@@ -126,10 +128,12 @@ function ajax (input) {
     try { req.responseJSON = JSON.parse(req.responseText) } catch (err) { req.responseJSON = false }
     dispatchEvent(input, 'input.ajax', req)
   }
-  ajax.timer = setTimeout(() => { // Debounce next request 500 milliseconds
+  ajax.timer = setTimeout(() => {
     if (!input.value) return // Abort if input is empty
-    req.open('GET', url.replace('{{value}}', window.encodeURIComponent(input.value)), true)
-    req.setRequestHeader('X-Requested-With', 'XMLHttpRequest') // https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Requested-With
-    req.send()
-  }, AJAX_DEBOUNCE)
+    if (dispatchEvent(input, 'input.ajax.beforeSend', req)) {
```

Comment:
Maybe a stupid question, but what if in the `ajax.beforeSend` event handler the user executes `req.send()`/`req.open()`? Will that trigger an error when we call `req.send()`/`req.open()`?  Should we handle that?